### DISCUSSION TITLE
How does LightningLite handle the grad scaler state dict of torch.amp?

### DISCUSSION THREAD

**Discussion by hiyyg** [Lightning-AI/pytorch-lightning#10727] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Is the grad scaler included in the `model.state_dict()` after steup by `model, optimizer = self.setup(model, optimizer)`?

If not, how can I save and load the state of the grad scaler for resume?

**Comment by tchaton** [comment, marked as answer]
Hey @hiyyg.

You can access the scale through the precision plugin as follows:

```py
self._precision_plugin.scaler
```

I believe you could get the state and reload it manually.

Best,
T.C

**Reply by hiyyg** [reply]
OK. I will try that.
